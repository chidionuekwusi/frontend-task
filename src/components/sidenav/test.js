import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import SideNav from "./index";
import NotesContext from "../../contexts/notes";

test("side nav renders list of notes", () => {
  const component = renderer.create(
    <NotesContext.Provider
      value={{
        get: () => {
          return [
            { title: "Fake", content: "Content", id: "1" },
            { title: "Fake 2", content: "Content 2", id: "2" }
          ];
        }
      }}
    >
      <MemoryRouter>
        <SideNav />
      </MemoryRouter>
    </NotesContext.Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
