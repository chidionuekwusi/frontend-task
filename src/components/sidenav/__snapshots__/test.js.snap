// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side nav renders list of notes 1`] = `
<ul>
  <li>
    <a
      href="/note/1"
      onClick={[Function]}
    >
      Fake
    </a>
  </li>
  <li>
    <a
      href="/note/2"
      onClick={[Function]}
    >
      Fake 2
    </a>
  </li>
</ul>
`;
